src/shogun/machine/gp/EPInferenceMethod.cpp

```diff
@@ -447,7 +447,8 @@ SGVector<float64_t> CEPInferenceMethod::get_derivative_wrt_inference_method(
 	SGVector<float64_t> result(1);
 
 	// compute derivative wrt kernel scale: dnlZ=-sum(F.*K*scale*2)/2
-	result[0]=-(eigen_F.cwiseProduct(eigen_K)*m_scale*2.0).sum()/2.0;
+	result[0]=-(eigen_F.cwiseProduct(eigen_K)).sum();
+	result[0]*=CMath::exp(m_log_scale*2.0);
```

Comment:
I agree, but the get_scale thing is much more easy to read. Doesnt matter though